### PR TITLE
Feat: fetch and display user from the backend

### LINKED ISSUE #94: Fetch and display user form the backend.
### Objective

The objective is to show the currently logged in user, and making sure they stays logged in using the refresh token.

### Description

* After the user logs in, capture the token ID and all the data that is sent back from the backend
* store the user on the async storage 
* store the **access_token** on the async storage 
* store the **refresh_token** on the async storage 
* Display the user information on the front end. 
* Add a function to refresh the access token. 

### Acceptance criteria

* User requests go through
* User information is displayed from the backend.
* The user is refreshed using the refresh token
* No unnecessary changes have been introduced in the code.

### PR DESCRIPTION
**Pull request by serapieTuyishime**
Closes #94 

### Acceptance criteria.

- [ ] User requests go through
- [ ] User information is displayed from the backend.
- [ ] The user is refreshed using the refresh token
- [ ] No unnecessary changes have been introduced in the code.

### CODE REVIEW

File: components/cards/UserAvailability.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import CustomText from "@/components/ui/CustomText"
 import { theme } from "@/theme"
 
 interface UserAvailabilityProps extends ViewProps {
-  fullName: string
+  fullName: string | null
```
Comment (igorntwari) [inline components/cards/UserAvailability.tsx:6]:
```suggestion
  fullName: string
```

File: components/cards/UserStatus.tsx [hunk 2]
```diff
@@ -11,12 +10,13 @@ import Animated, {
   withTiming,
 } from "react-native-reanimated"
 import UserAvailability from "@/components/cards/UserAvailability"
+import { User } from "@/types"
```
Comment (igorntwari) [inline components/cards/UserStatus.tsx:13]: This is not necessary

File: components/cards/UserStatus.tsx [hunk 3]
```diff
@@ -11,12 +10,13 @@ import Animated, {
...
 
 const MAX_VISIBLE_FRIENDS = 3
 
 interface UserStatusProps extends ViewStyle {
   friends: User[] | []
-  user: User
+  user: User | null
```
Comment (igorntwari) [inline components/cards/UserStatus.tsx:19]:
```suggestion
  user: User
```

File: components/cards/UserStatus.tsx [hunk 4]
```diff
@@ -32,7 +32,11 @@ export default function UserStatus({
   const remainingCount = Math.max(friends.length - MAX_VISIBLE_FRIENDS, 0)
 
   const fullFriendsList = visibleFriends
-    .map((friend) => `${friend.name}`)
+    .map((friend) => {
```
Comment (igorntwari) [inline components/cards/UserStatus.tsx:35]: We no longer have have firstName and lastName just use name

File: contexts/AuthContext.tsx [hunk 5]
```diff
@@ -56,15 +51,28 @@ export const AuthProvider: React.FC<{ children: ReactNode }> = ({
 
       if (isSuccessResponse(response)) {
         const idToken = response.data.idToken ?? ""
-        const user = response.data.user
-        await AsyncStorage.setItem("@Auth:token", idToken)
-        await AsyncStorage.setItem("@Auth:user", JSON.stringify(user))
+        const user_from_auth = response.data.user
```
Comment (igorntwari) [inline contexts/AuthContext.tsx:54]: can you change this name to something like this ```authenticatedUser```,```currentUser```,```loggedInUser``` or any other convenient name

File: contexts/AuthContext.tsx [hunk 6]
```diff
@@ -56,15 +51,28 @@ export const AuthProvider: React.FC<{ children: ReactNode }> = ({
...
+          activity: "Hangout",
+        }
         setUser(user)
 
-        await axios.post(`${process.env.API_BASE_URL}/auth/google-signin`, {
-          token: idToken,
-          platform: Platform.OS,
-        })
+        const { data } = await axios.post(
+          `${process.env.API_BASE_URL}/api/auth/google-signin`,
```
Comment (igorntwari) [inline contexts/AuthContext.tsx:65]: the ```API_BASE_URL``` includes the ```api``` part you don't need to  include it again

File: service/index.ts [hunk 7]
```diff
@@ -34,8 +34,22 @@ api.interceptors.response.use(
     return response
   },
   async (error: AxiosError) => {
-    if (error.response?.status === 401) {
-      // Handle unauthorized errors
+    const refreshToken = await AsyncStorage.getItem("@Auth:refreshToken")
+    if (error.response?.status === 401 && refreshToken) {
+      const url = `${process.env.API_BASE_URL}/api/auth/refresh-token`
```
Comment (igorntwari) [inline service/index.ts:39]: even here

File: service/index.ts [hunk 8]
```diff
@@ -34,8 +34,22 @@ api.interceptors.response.use(
...
+      }
+      const { data } = await axios.post(url, body)
+
+      const accessToken = data.access_token
+      const new_refreshToken = data.refresh_token
+      await AsyncStorage.setItem("@Auth:accessToken", accessToken)
+      await AsyncStorage.setItem("@Auth:refreshToken", new_refreshToken)
+
+      error.response.config.headers["Authorization"] = "Bearer " + accessToken
+      console.log("============Successfully refresh token!=============")
```
Comment (igorntwari) [inline service/index.ts:51]: remove these console

File: types/index.ts [hunk 9]
```diff
@@ -38,7 +38,7 @@ export type Theme = {
 export interface User {
   id: string
   username: string
-  name: string
+  name: string | null
```
Comment (igorntwari) [inline types/index.ts:41]: instead of making the name ```null``` can you inspect your code base try to find what makes it null ?

**Review by igorntwari — CHANGES_REQUESTED** [review]

File: types/index.ts (hunk 9 above)
Comment (serapieTuyishime) [reply types/index.ts:41]: It was the user from google, however I added a fix.

**Review by igorntwari — APPROVED** [review]

**Review by mhenriette — APPROVED** [review]

**Pull request merged**